### ISSUE TITLE
Touch event on a link issue

### ISSUE THREAD

**Issue by bullgare** [angular/angular.js#5307]
It's in AngularJS 1.2.3.
This is only happening on touch devices or in Google Chrome with touch event emulation enabled.
This link works (redirects to the url):
```<a href="#/search">search</a>```
And this link doesn't (no redirect occurs on tap):
```<a href="#/search" ng-click="v = 1">search</a>```
Click handler invoked and after that something strange happens inside of 
```element.on('touchend', function(event)``` (angular-touch.js:385). 
Somehow it decides to prevent event by ```calling preventGhostClick(x, y);```
It's really sad because it blocked our big mobile app to use AngularJS 1.2.3 in production. We're still on 1.0.8.

**Comment by ostux** [comment]
+1 this bug still exist....

**Comment by MaciejHadam** [comment]
default behaviour of link is broken, +1 to fix this bug

**Comment by stianlp** [comment]
+1

**Comment by xinsight** [comment]
+1
Another workaround is to add the following into your ng-click handler.

```
// <a ng-click="hideMenu($event)" ng-href="#/settings">
$scope.hideMenu = function (event) {
    // ...
    var href = angular.element(event.target).attr('href');
    if (event.type === 'touchend' && href) {
        href = href.substring(1); // remove initial #
        $location.path(href);
    }
};
```

**Comment by WraithKenny** [comment]
It'd be nice if we can get a response on wether this is a bug or a feature...I think it's a bug, but I'm new to Angular...

Is it recommended to not use ngHref when also using ngClick? If it is, I'll deal with it I guess, otherwise, it'd be nice to have this confirmed as a bug, and we can expect a bug fix sometime in the future.

**Comment by pimlinders** [comment]
+1

**Comment by mping** [comment]
+1 Same bug here.

**Comment by fishermanswharff** [comment]
+1

**Comment by arifeng** [comment]
+1

**Comment by tspore** [comment]
+1

**Comment by eantze** [comment]
+1

**Comment by maseh87** [comment]
+1

**Comment by derekprior** [comment]
Please stop with the :+1:ing. It's not adding anything to the discussion of the actual issue and only emails people hwo are watching this for actual progress unnecessarily.

**Comment by WraithKenny** [comment]
Better, is to wait on Angular 2 and hope that doesn't have this bug. Just unsubscribe because there will probably never be any progress on this bug in version 1.

**Comment by asfaltboy** [comment]
There seems to be [some code that ensures the tap did not "end" too far away from origin](https://github.com/angular/angular.js/blob/d077966ff1ac18262f4615ff1a533db24d4432a7/src/ngTouch/directive/ngClick.js#L84-L91), called clickbuster.

This may be related to the CLICKBUSTER_THRESHOLD, which is set to 25px (too small for chubby thumbs?).

**Comment by David-Tsui** [comment]
still exists, I think

**Comment by Narretz** [comment]
The touch override from ngTouch has been disabled by default since 1.5.0, so I doubt this issue still exists in this version. It may be an issue with 3rd party touch handlers. There's also a point to be made that having a click handler on a link is bad for accessibility.

**Comment by rendisek70** [comment]
Dne 27. 10. 2016 8:12 napsal uživatel "David-Tsui" <notifications@github.com

**Comment by Narretz** [comment]
I am going to lock this issue, as I don't think we can lead a constructive discussion here. Please open a new issue with a clear description and a runnable example if you have problems with this in the current Angular version (1.5.8).
